test/integration/metricspullthrough/metricspullthrough_test.go

```diff
@@ -0,0 +1,167 @@
+		}
+
+		switch req {
+		case "GET /v2/":
+			w.Write([]byte(`{}`))
+		case "GET /v2/remoteimage/manifests/latest":
+			mediaType := "application/vnd.docker.distribution.manifest.v2+json"
+			w.Header().Set("Content-Type", mediaType)
+			_ = json.NewEncoder(w).Encode(map[string]interface{}{
+				"schemaVersion": 2,
+				"mediaType":     mediaType,
+				"config": map[string]interface{}{
+					"mediaType": "application/vnd.docker.container.image.v1+json",
+					"size":      len(config),
+					"digest":    configDigest.String(),
+				},
+				"layers": []map[string]interface{}{
+					{
+						"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
+						"size":      len(foo),
+						"digest":    fooDigest.String(),
+					},
+				},
+			})
+		case "GET /v2/remoteimage/blobs/" + configDigest.String():
+			w.Write([]byte(config))
+		case "HEAD /v2/remoteimage/blobs/" + fooDigest.String():
+			w.Header().Set("Content-Length", fmt.Sprintf("%d", len(foo)))
+			w.WriteHeader(http.StatusOK)
+		case "GET /v2/remoteimage/blobs/" + fooDigest.String():
+			w.Write([]byte(foo))
+		default:
+			t.Errorf("error: remote registry got unexpected request %s: %#+v", req, r)
+		}
+	}))
+	defer ts.Close()
+
+	imageClient := imageclientv1.NewForConfigOrDie(master.AdminKubeConfig())
+
+	isi, err := imageClient.ImageStreamImports(testproject.Name).Create(&imageapiv1.ImageStreamImport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: teststreamName,
+		},
+		Spec: imageapiv1.ImageStreamImportSpec{
+			Import: true,
+			Images: []imageapiv1.ImageImportSpec{
+				{
+					From: corev1.ObjectReference{
+						Kind: "DockerImage",
+						Name: fmt.Sprintf("%s/remoteimage:latest", ts.URL.Host),
+					},
+					ImportPolicy: imageapiv1.TagImportPolicy{
+						Insecure: true,
+					},
+				},
+			},
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	teststream, err := imageClient.ImageStreams(testproject.Name).Get(teststreamName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(teststream.Status.Tags) == 0 {
+		t.Fatalf("failed to import image: %#+v %#+v", isi, teststream)
+	}
+
+	remoteRegistryRequiresAuth = true
+
+	registry := master.StartRegistry(t, testframework.DisableMirroring{}, testframework.EnableMetrics{Secret: "MetricsSecret"})
+	defer registry.Close()
+
+	repo := registry.Repository(testproject.Name, teststream.Name, testuser)
+
+	ctx := context.Background()
+
+	_, err = repo.Blobs(ctx).Get(ctx, fooDigest)
+	if err != distribution.ErrBlobUnknown {
+		t.Fatal(err)
+	}
+
+	req, err := http.NewRequest("GET", registry.BaseURL()+"/extensions/v2/metrics", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.Header.Set("Authorization", "Bearer MetricsSecret")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer resp.Body.Close()
+
+	found := false
+	r := bufio.NewReader(resp.Body)
+	for {
+		line, err := r.ReadString('\n')
+		if strings.HasPrefix(line, "imageregistry_pullthrough_repository_errors_total{") {
```

Comment:
Have you considered using the client library e.g.: https://github.com/prometheus/client_golang/issues/194#issuecomment-194319793 ?